addons/pos_restaurant/static/src/app/screens/floor_screen/floor_screen.js

```diff
@@ -530,6 +529,25 @@ export class FloorScreen extends Component {
         const table = await this.pos.data.create("restaurant.table", [newTableData]);
         return table[0];
     }
+    async unMergeTable(table) {
+        const mainTable = this.pos.getMainTable(table);
+        const mainOrder = this.pos.getActiveOrdersOnTable(mainTable)?.[0];
+        const orderToRestore = this.pos.models["pos.order"].find(
+            (o) =>
+                (o.table_id?.id === table.id || o.origin_table_id?.id === table.id) && !o.finalized
+        );
```

Comment:
```suggestion
        const orderToRestore = table["<-pos.order.origin_table_id"].find((o) => !o.finalized);
```

Reply:
this will work only when destination table is empty 
because only at that time we update the table_id of order 

but when both tables have orders then it will not work 

```
const orderToRestore =
            table["<-pos.order.origin_table_id"].find((o) => !o.finalized) ||
            table["<-pos.order.table_id"].find((o) => !o.finalized);
```
in that case we need to use this